### ISSUE TITLE
Consider deprecating `join_abcd_scenario` and `summarize_*`

### ISSUE THREAD

**Issue by jdhoffa** [RMI-PACTA/r2dii.analysis#468]
These functions are currently exported, but I have no idea if anybody uses them. 
To some extent, it feels like we are committing ourselves to a maintenance burden for a decision made long ago. 

I am curious if @jacobvjk thinks they are actively used (beyond when they are called in the body of `target_*` functions)
Curious also how @AlexAxthelm and @cjyetman feel about this (not the details, but in general the process of deprecating a previously exported function)

It feels like probably a bad idea/ bad form, but want your opinions here.

**Comment by cjyetman** [comment]
I think that's what "deprecation" is intended for, no?

**Comment by AlexAxthelm** [comment]
Yeah, Let's support them for another few months, and then internalize them.

**Comment by jdhoffa** [comment]
Maybe more concretely, my question is:
* Should I include, in the next release that I push to CRAN, a warning message explaining that we plan to deprecate these functions? (and maybe even a link to making an issue to this repo in case people wish for us to not deprecate them)?

I think this may be an appropriate strategy here

**Comment by AlexAxthelm** [comment]
I think this is exactly what `deprecate_soft()` is for, no? Anyone who calls the function directly sees the deprecation warning, anyone who uses it indirectly gets the expected result, some time later, we'll unexport them.

**Comment by jdhoffa** [comment]
Sounds perfect!

**Comment by cjyetman** [comment]
@jdhoffa and @jacobvjk noting that the documentation of these functions in [pacta.loanbook](https://github.com/RMI-PACTA/pacta.loanbook) somewhat increases the urgency of deciding whether or not to deprecate these functions

**Comment by jacobvjk** [comment]
I would personally favor not supporting these functions as exported user-facing functions anymore, because we never use them in any standard PACTA use case. But I can't say if there are banks users who consider these functions fundamental to their using PACTA or not. Maybe we announce the planned deprecation and if no one complains we can proceed with next steps in the next release?

**Comment by jdhoffa** [comment]
I'm happy to move forward with deprecating these functions.
To begin the deprecation process, we need one release where they still are available, but warn that we plan to deprecate them.

**Comment by jacobvjk** [comment]
> I'm happy to move forward with deprecating these functions. To begin the deprecation process, we need one release where they still are available, but warn that we plan to deprecate them.

In my understanding, the `deprecate_soft()` step means that there will be two more steps. This one indicated only internally/to developers that these function will be deprecated. It does not throw a warning yet. Next step would be `deprecate_warn()`, warning users explicitly. And then finally, `deprecate_stop()`... we can theoretically jump the initial step.. I don't have a strong view on that
